### ISSUE TITLE
[Feature] Support for named external blocks

### ISSUE THREAD

**Issue by SirLynix** [NazaraEngine/ShaderLang#50]
Currently, external blocks brings values in the global scope:
```rs
struct MaterialData
{
    baseColor: vec3[f32]
}

struct InstanceData
{
    worldMatrix: mat4[f32]
}

[auto_binding]
external
{
    viewerData: uniform[ViewerData]
}

[auto_binding]
external
{
    matData: uniform[MaterialData],
    baseColorMap: sampler2D[f32]
}

// matData, baseColorMap and viewerData are usable here
```

it would be a good idea to support named external blocks, such as:
```rs
[auto_binding]
external Viewer
{
    data: uniform[ViewerData] // Available as Viewer.Data
}

[auto_binding]
external Material
{
    data: uniform[MaterialData], // available as Material.data
    baseColorMap: sampler2D[f32] // available as Material.baseColorMap
}
```

**Comment by Kbz-8** [comment]
Wouldn't it be better to just have namespaces kind of like C++ ones so that functions and structures could also have namespaces?

**Comment by Kbz-8** [comment]
yes, but I'd find it annoying to make a new file just to create a namespace

**Comment by SirLynix** [comment]
I think we can keep named external for now and maybe add namespace in the future, however before namespace are added I'd like to add support for named module imports.

```rs
import Foo from Bar;

// Bar.Foo is available under the name Foo
```

maybe allow
```rs
import Bar.Foo from Bar;

// Bar.Foo is available
```
and
```rs
import Bar.* from Bar;

// everything in Bar is available under the Bar. namespace
```

I'm not fan of repeating the module name though

**Comment by SirLynix** [comment]
Implemented in #52.
